### ISSUE TITLE
Ombi does not allow requesting episodes that have been requested previously (or being already monitored in Sonarr)

### ISSUE THREAD

**Issue by softworkz** [tidusjar/Ombi#1952]
#### Ombi build Version:

c3c85b0ae09f04e4ef00736c6025fea0a547a88b (current as of writing)

#### Update Branch:

DotNetCore

#### Media Sever:

Irrelevant

#### Media Server Version:

Irrelevant

#### Operating System:

Irrelevant

#### Ombi Applicable Logs (from `/logs/` directory or the Admin page):

No logs required

#### Problem Description:

Scenario 1:

- You create a request for a number of episodes
- The request is auto-approved
- Episodes are added to Sonarr and marked as monitored but either no search has been performed in Sonarr or there weren't any results (could be due to various reasons, e.g. quality, language etc)
- Then you delete the request in OMBI
- After a while, you decide to request those episodes again via OMBI, but OMBI is displaying those episodes as "Processing Request" 
- This makes it impossible to request those episodes again!

Scenario 2:

- You have a pre-existing series in Sonarr with some episodes in the "monitored" state but not downloaded
- This means that those episodes will only get downloaded once they appear in the feed that Sonarr is watching
- You have never made any requests for this series via OMBI
- It could easily be possible though, that the episodes would be found when performing a search with Sonarr
- Now you search for this series in OMBI
- But in OMBI you can't request any of the monitored episodes because they are displayed as "Processing Request"

**Comment by softworkz** [comment]
I believe that the problem is here:

https://github.com/tidusjar/Ombi/blob/a8bb352c345e5f7292cf44a76db438d66ae887d0/src/Ombi.Core/Rule/Rules/SonarrCacheRule.cs#L75

and here:

https://github.com/tidusjar/Ombi/blob/d3502857f1ef8493f395869844fa7df5768fab51/src/Ombi/ClientApp/app/search/seriesinformation.component.html#L59

When an episode is being monitored in Sonarr, the _Approved_ property is set to true, even if it hasn't been requested via OMBI, and in the html template line, there's also no checking for the _Requested_ property. 

I don't know what would be the best solution here, but this is how it comes to that situation...

**Comment by softworkz** [comment]
I hope you don't mind me adding some personal comment here: I like OMBI, I think it's cool in what it is doing and how it is doing that. Before I had come accross OMBI (just few weeks ago), I always had in mind to do something similar. 

But there is one single issue that I find really annoying - the "Processing Request" state.
I'm only mentioning it here because this bug/issue is closely related.

I think the item state could be implemented in a much better way:

- "Processing Request" should only be shown until there's not feedback from Sonarr/Radarr
- When an item is already monitored in S/R (and not requested via OMBI), the state should be displayed as "Monitored", but it should still be possible to create a request for the item in order to force a search

Even more important IMO would be this:
- When you create a request via OMBI currently, this remains to be a shot into the dark
- The item will show up as "Processing Request" while this could mean anything between: "Request will be available in a few minutes" and "might take a number of months"
I think that we could provide much better information to the user in this case like:
- Found. Queued for download
- Download failed. Waiting for alternate sources.
- Available files not matching quality requirements
- No files available yet

Bottom line is: When you request something, one really important question is, whether you just need to wait for a short time until it is downloaded or if it could take weeks or even months until it is there...
If OMBI could provide this information it would be perfect :-) :-)

**Comment by tidusjar** [comment]
Hey,

Thanks for putting so much detail into this! Amazing!

So the reason why you cannot re-request something that is montiored in Sonarr is that, it wouldn't really do much, there wouldn't really be much point in it in my opinion.

Ok, so I like your point with adding the new state "Monitoring", Might be a bit ambiguous to the user, but it's better than what we currently have.

Now with the information about the actual download, I don't really want that sort of information inside Ombi, plus i'd have to poll S/R for that information every x minutes.

**Comment by softworkz** [comment]
> Now with the information about the actual download, I don't really want that sort of information inside Ombi, plus i'd have to poll S/R for that information every x minutes.

Yeah OK, this might be really a bit too much. The primary purpose is really to cover situations like when you create a request for a movie or episodes e.g. in the afternoon, to get an indication whether you can expect being able to watch it in the evening or if you would need to wait until it becomes available..

Regarding the "Monitoring" state - I agree! Initially I had written "Monitored by Sonarr" which is not a good state semantically. But well - the client could construct it this way...

**Comment by softworkz** [comment]
> So the reason why you cannot re-request something that is montiored in Sonarr is that, it wouldn't really do much, there wouldn't really be much point in it in my opinion.

The significant point here is caused by the way Sonarr is working in combination with Newznab indexers. With most Newznab indexers there are two different methods to acquire content data:

1. Recent items feed
2. Search

With the recent items feed, you always get a list about the n items that have just recently been uploaded, while the search method allows you to search the indexer's complete database for a specific item.

This is important to understand how Sonarr is working:

- When an item is "monitored" by Sonarr, it means that Sonarr will regularly check the "Recent Items Feed" (1.) to see if it includes an appropriate release for that item
- But Sonarr will never(!) perform a search (2.) on its own (i.e. on regular schedule)

And that's what makes the difference: Creating a request via OMBI initiates a "Search" in Sonarr, no matter whether it's aleady monitored or not.

In a perfect world we could assume that for any item that is monitored in Sonarr, the search has been performed at least once - and in this case further searches could be pointless and it would be sufficient to follow the "Recent Items Feed" regularly.

But a lot of things can happen here and lead to situations where items are marked as monitored in Sonarr without any search performed.

Examples:

- You have changed your Sonarr configuration like quality profiles etc.
- You changed your list of indexers
- You added existing series to Sonarr
- You created a request via OMBI and deleted it before a search could be performed (either quickly or maybe also because you exceeded your indexer's quota)
- You simply marked a series, season, episode as monitored in the Sonarr GUI without performing a search
- Indexer search failed (various reasons possible) when adding an item. Item is added as monitored without having been searched
(I'm sure ths list isn't even complete)

Bottom line is: When an item is in "monitored" state in Sonarr and hasn't been downloaded yet, you can't take it for granted that the item wouldn't be available when performing a search. 
And this is why it is very important to be able to create requests for (in Sonarr) monitored items via OMBI: To initiate a search for the item.

**Comment by tidusjar** [comment]
Very valid points. I agree with your suggestions for changes!

**Comment by ceptonit** [comment]
I can't seem to be able to request a movie that has previously been requested, even if I remove the movie manually from Radarr.
Makes sense to me not being able to request movies if they're already in Radarr, but if removed, they should be able to be requested again imo.

Didn't know if I had to make a new thread as this is the same issue as OP, instead in Radarr.

**Comment by tidusjar** [comment]
@ceptonit it's not the same issue. What you are seeing is the way ombi currently works. In the future i plan to change this. We do not check radarr to see if stuff has been removed.

**Comment by ceptonit** [comment]
@tidusjar gotcha. Should I open a new issue so it can be tracked?

**Comment by tidusjar** [comment]
It's not an issue. I'll change it when I have time

**Comment by stevephillips** [comment]
Just to add my input, I would like this added just to receive an email notification when the movie is marked as available. Multiple users of my plex server might all want the same movie and wish to be notified, so in this instance requesting the same movie from multiple users makes sense.

**Comment by enyo** [comment]
> Just to add my input, I would like this added just to receive an email notification when the movie is marked as available. Multiple users of my plex server might all want the same movie and wish to be notified, so in this instance requesting the same movie from multiple users makes sense.

I agree with @TangoMikeOscar. I think that this is not exactly the same issue as described here so maybe it makes sense to open a new one, but it would be great if multiple people could request the same movies/shows to receive notifications when they are available.

**Comment by kewlrobd** [comment]
I would also like to see multiple people be able to receive a notification on a requested title.  Maybe make the requested button clickable.  Clicking it would add that user to the notifications on that title.  

In regards to the requester knowing the status of a request I include a link to my utorrent Web UI in the approved notification email.  This at least gives them some idea if the title was readily available or not.  I know not everyone's setup will allow for this but it works really well for me.

**Comment by kewlrobd** [comment]
@tidusjar I apologize. I was simply directly addressing the last two posts in this thread.

**Comment by Wdavery** [comment]
Why is this closed? This still seems relevant as it has not been addressed at all.

**Comment by MLWeber** [comment]
I too would like to have ombi initiate a search upon request, even if the item is already monitored.

**Comment by bolagnaise** [comment]
I agree, this needs to be a feature!
